### ISSUE TITLE
Accessing SUSystemProfiler output to present to users

### ISSUE THREAD

**Issue by pilky** [sparkle-project/Sparkle#1616]
Hey, I'm wanting to give users a way to see what data is set to my server during an update. This is possible during the initial UI, but if someone clicks through that UI or if they want to double check what data is sent (which may have changed since first enabling updates) there is no way to do that.

To fix this I want to present the data from SUSystemProfiler to the user. Unfortunately this isn't public (presumably because it requires SUHost which also isn't public). Would anyone have any objections to adding a `systemProfileArray` property to SPUUpdater that basically returns `[SUSystemProfiler systemProfileArrayForHost:self.host]`? 

I'm probably going to do that myself regardless, this is more a question of whether you would be open to me creating a PR afterwards to get it into the 2.x branch?

**Comment by kornelski** [comment]
Adding that method seems fine.

**Comment by pilky** [comment]
I've added a PR for this change (and a few others) at https://github.com/sparkle-project/Sparkle/pull/1690. This is my first Sparkle PR so hopefully I've got everything ok but please let me know if there are any questions or any changes needed

**Comment by pierswalter** [comment]
I believe that this issue may be resolved with my PR #1687 which had been merged on Nob 27, 2020.

**Comment by pilky** [comment]
@pierswalter Just as a heads up, this will conflict with my similar changes on the 2.x branch. I've named my property `systemProfileArray`, though it works a bit differently. Rather than storing the profile it generates it whenever asked. It also hooks into the additional filtering of profile keys I added to the delegate. I also switched all other locations over to calling the property rather than SUSystemProfiler directly to ensure this filtering is correctly applied everywhere
